test/parallel/test-process-chdir.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+assert.notStrictEqual(process.cwd(), __dirname);
+process.chdir(__dirname);
+assert.strictEqual(process.cwd(), __dirname);
+
+const dir = path.resolve(common.tmpDir,
+    'weird \uc3a4\uc3ab\uc3af characters \u00e1\u00e2\u00e3');
+
+// Make sure that the tmp directory is clean
+common.refreshTmpDir();
```

Comment:
I think this is effectively unsafe after using `common.tmpDir`?

Isn't it possible that your tmpDir is different then?